sw/device/silicon_creator/rom/data/rom_testplan.hjson

```diff
@@ -134,400 +121,1008 @@
+            |         On          |   Virtual  |     B     |  No   |
+            |         Invalid     |   Virtual  |     A     |  Yes  |
+            |         Invalid     |   Virtual  |     B     |  Yes  |
+            '''
+      tags: ["rom", "verilator", "dv", "fpga", "silicon"]
+      milestone: V2
+      tests: []
+    }
+
+    {
+      name: rom_e2e_debug
+      desc: '''Verify that ROM can be debugged in appropriate life cycle states.
+
+            `CREATOR_SW_CFG_ROM_EXEC_EN` should be set to `0`.
+
+            - Verify that ROM can be debugged in TEST, DEV, and RMA life cycle states.
+              - Connect a debugger and verify that ROM halts very early in `rom_start.S`.
+              - Step over the `wfi` instruction.
+              - Verify that ROM fails to boot with `BFV:0142500d`.
             '''
       tags: ["rom", "verilator", "dv", "fpga", "silicon"]
       milestone: V2
       tests: []
     }
+
     {
-      name: rom_e2e_bootstrap_disabled
-      desc: '''Verify device does not enter bootstrap mode when strapping pins are set correctly, but bootstrap
-            is disabled in the OTP.
+      name: rom_e2e_debug_disallowed_in_prod
+      desc: '''Verify that ROM can be debugged in appropriate life cycle states.
 
-            - Test runner loads an OTP image that has bootstrapping mode disabled.
-            - Set the strapping pins to the correct levels for bootstrapping.
-            - Attempt to boot the chip.
-            - Verify that the chip does not enter bootstrap mode.
+            `CREATOR_SW_CFG_ROM_EXEC_EN` should be set to `0`.
+
+            - Verify that ROM cannot be debugged in PROD and PROD_END.
             '''
       tags: ["rom", "verilator", "dv", "fpga", "silicon"]
       milestone: V2
       tests: []
     }
 
-    // Bootstrap mode
     {
-      name: rom_e2e_bootstrap_config
-      desc: '''Verify that JEDEC ID and SFDP are configured.
+      name: rom_e2e_jtag_inject
+      desc: '''Verify that a program can be injected into SRAM via JTAG and executed.
+
+            `CREATOR_SW_CFG_ROM_EXEC_EN` should be set to `0`.
 
-            - Test runner configures chip to enter bootstrap mode.
-            - In bootstrap mode, read JEDEC Device ID
-            - Verify that it matches the lowRISC JEDEC ID.
-            - SW reads SFDP and verifies that it is correct.
+            - For TEST, DEV, and RMA:
+              - Connect a debugger.
+              - Follow the steps
+                [here](https://docs.google.com/document/d/1i6V3tFTv0UhWeM-IGCxgxkC07Bo_zpbgr6u1KE1Rehs/edit?resourcekey=0-L-XaP9QeLAkEsp8najVkUQ#heading=h.m4f6p97t15j8)
+                and execute `//sw/device/examples/sram_program` binary from SRAM.
+              - Verify that chip sends the expected message over UART.
+
+            See this
+            [doc](https://docs.google.com/document/d/1i6V3tFTv0UhWeM-IGCxgxkC07Bo_zpbgr6u1KE1Rehs/edit?resourcekey=0-L-XaP9QeLAkEsp8najVkUQ)
```

Comment:
Same here regarding moving this to markdown on the website?

Reply:
Thanks! but I'm not sure about this one as is -- maybe a condensed version? I'll add to the issue to evaluate later.